### ISSUE TITLE
Document and raise better exceptions about backend loading

### ISSUE THREAD

**Issue by relativistic** [ioam/holoviews#999]
When I try to use the %%output cell magic to switch to the bokeh backend, I get the error "Error: list index out of range".

![screen shot 2016-11-30 at 11 10 12 am](https://cloud.githubusercontent.com/assets/371039/20767041/9c180ee2-b6ed-11e6-886b-5f940fb06491.png)

**Comment by jlstevens** [comment]
I believe you just need to load the bokehJS. At the top of the notebook, try:

```
import holoviews as hv
hv.notebook_extension('matplotlib', 'bokeh')
```

Then it should work as long as bokeh is installed.

**Comment by philippjfr** [comment]
Correct, definitely needs a better exception though. There's a few issues like this now that the plotting backends aren't imported by default, so we should catalogue them and make sure to raise better exceptions.

**Comment by relativistic** [comment]
Ah, yes, that fixed it. Thanks!

It might also be a good idea to be explicit this is required "Bokeh Backend" page of the docs, where this is described. I had assumed from the language there that you could use the cell output magic after only loading the matplotlib option in the hv.notebook_extension. Probably obvious to those who are more familiar with the way this kind of thing works, but it wasn't for me.

**Comment by philippjfr** [comment]
Reopening and repurposing this issue.

**Comment by jlstevens** [comment]
Should be easy enough to fix so I'm assigning this as something to fix by 1.7.

**Comment by philippjfr** [comment]
The output magic error message was improved, closing.
